src/lib/identity/okta.ts

```diff
@@ -0,0 +1,79 @@
+import type {
+	AccessTokenClaims,
+	CustomClaims,
+	IdentityAuthState,
+} from '@guardian/identity-auth';
+import { IdentityAuth } from '@guardian/identity-auth';
+import config from 'lib/config';
+import type { Stage } from '../../types/global';
+
+// the `id_token.profile.theguardian` scope is used to get custom claims
+type CustomIdTokenClaims = CustomClaims & {
+	email: string;
+	google_tag_id: string;
+};
+
+const getStage = () => {
+	return window.guardian.config.page.stage;
+};
+
+const getIssuer = (stage: Stage) =>
+	stage === 'PROD'
+		? 'https://profile.theguardian.com/oauth2/aus3xgj525jYQRowl417'
+		: 'https://profile.code.dev-theguardian.com/oauth2/aus3v9gla95Toj0EE0x7';
+
+const getClientId = (stage: Stage) =>
+	stage === 'PROD' ? '0oa79m1fmgzrtaHc1417' : '0oa53x6k5wGYXOGzm0x7';
+
+const getRedirectUri = (stage: Stage) => {
+	switch (stage) {
+		case 'PROD':
+			return 'https://www.theguardian.com/';
+		case 'CODE':
+			return 'https://m.code.dev-theguardian.com/';
+		case 'DEV':
+		default:
+			return 'http://localhost:9000/';
+	}
+};
+
+let identityAuth:
+	| IdentityAuth<AccessTokenClaims, CustomIdTokenClaims>
+	| undefined;
+
+function getIdentityAuth() {
+	if (identityAuth === undefined) {
+		const stage: Stage = getStage();
+
+		identityAuth = new IdentityAuth<AccessTokenClaims, CustomIdTokenClaims>(
```

Comment:
Curious as to what happens if there are 2 instances of IdentityAuth on a page (1 from Commercial, 1 from Frontend/DCR) 🤔 Will they both be trying to refresh the token? Do we maybe need to expose a single instance of IdentityAuth on the window? Maybe a question for Identity...

Reply:
We'd want to share one instance of IdentityAuth between all js on the page! This will cause issues especially around token refreshing, as they'll all try to refresh at the same time.

We only have a limited number of API calls we can make to Okta per minute, so anything that will reduce this will be helpful.

Reply:
How is it shared between DCR and Frontend? can we use the same method to share it?

Reply:
After discussion with @AshCorr we're going to use the logic from this PR: https://github.com/guardian/frontend/pull/26469

Either commercial or frontend will be able to initialise identityAuth and add it to the window when needed. The repo who looks for the identityAuth code second will then be able to use the object on the window, preventing multiple instances of identityAuth running at the same time.

We'll be looking to abstract this logic out into a separate library so that we don't need to worry about manually keeping frontend and commercial in sync with each other.

Reply:
Would we need this instantiation method on DCR too? aiui the instantiation of auth state on Frontend would only set it for Frontend rendered pages.

Reply:
We would need it for DCR too